pgjdbc/src/main/java/org/postgresql/copy/CopyManager.java

```diff
@@ -48,26 +48,24 @@ public CopyManager(BaseConnection connection) throws SQLException {
   }
 
   public CopyIn copyIn(String sql) throws SQLException {
-    CopyOperation op = null;
-    try {
-      op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    CopyOperation op = queryExecutor.startCopy(sql, connection.getAutoCommit());
+    if (op == null || op instanceof CopyIn) {
```

Comment:
I'm afraid, "exceptional way" was slower to execute here because code did use the exception object -> thus JVM had to create fully workable exception object, populate stacktrace, etc, etc.

On the other hand, I do not think one would be able to get measurable improvements via this change.

I would consider this as a mere code cleanup. "class cast exception" does not buy here much, so I think removing `cce` from "caused by" does clean up the resulting exception a bit.